File: examples/featured_products.py
"""Split an Amazon search page into featured (sponsored) and organic results.

Verified against the live ScrapingBee Amazon API on 2026-09-10.
Set SCRAPINGBEE_API_KEY in your environment before running.
"""

import os

import requests

BASE = "https://app.scrapingbee.com/api/v1"
KEY = os.environ["SCRAPINGBEE_API_KEY"]
HEADERS = {"Authorization": f"Bearer {KEY}"}


def search(query, pages=1, **params):
    """Amazon search results. 5 credits per page on a light request."""
    r = requests.get(
        f"{BASE}/amazon/search",
        headers=HEADERS,
        params={"query": query, "pages": pages, **params},
        timeout=60,
    )
    r.raise_for_status()
    return r.json()


def product(asin, **params):
    """Structured product detail. Takes 'query', not 'asin'."""
    r = requests.get(
        f"{BASE}/amazon/product",
        headers=HEADERS,
        params={"query": asin, **params},
        timeout=60,
    )
    r.raise_for_status()
    return r.json()


def pricing(asin, **params):
    """Seller offers. This endpoint takes 'asin', not 'query'."""
    r = requests.get(
        f"{BASE}/amazon/pricing",
        headers=HEADERS,
        params={"asin": asin, **params},
        timeout=60,
    )
    r.raise_for_status()
    return r.json()


if __name__ == "__main__":
    page = search("fitness tracker", sort_by="featured")
    products = page["products"]

    featured = [p for p in products if p["is_sponsored"]]
    organic = [p for p in products if not p["is_sponsored"]]

    print(f"{page['products_count']} products, {len(featured)} featured, {len(organic)} organic")

    for p in featured:
        print(f"  ad  slot {p['sponsored_position']}  {p['price']} {p['currency']}  {p['title'][:60]}")

    for p in organic[:5]:
        badge = "choice" if p["is_amazons_choice"] else ("bestseller" if p["best_seller"] else "")
        print(f"  org rank {p['organic_position']}  {p['price']} {p['currency']}  {badge}  {p['title'][:50]}")

    if organic:
        asin = organic[0]["asin"]
        detail = product(asin)
        print(f"\n{detail['brand']} / {detail['price']} {detail['currency']} / {detail['rating']} stars")
        for offer in pricing(asin)["pricing"]:
            print(f"  offer {offer['seller']}  {offer['price']}  {offer['condition']}")

File: examples/featured_products.js
// Split an Amazon search page into featured (sponsored) and organic results.
// Verified against the live ScrapingBee Amazon API on 2026-09-10.
// Set SCRAPINGBEE_API_KEY in your environment before running.

const axios = require('axios');

const BASE = 'https://app.scrapingbee.com/api/v1';
const KEY = process.env.SCRAPINGBEE_API_KEY;
const headers = { Authorization: `Bearer ${KEY}` };

async function search(query, params = {}) {
  const res = await axios.get(`${BASE}/amazon/search`, {
    headers,
    params: { query, ...params },
    timeout: 60000,
  });
  return res.data;
}

async function product(asin, params = {}) {
  // The product endpoint takes 'query', which accepts a 10 character ASIN.
  const res = await axios.get(`${BASE}/amazon/product`, {
    headers,
    params: { query: asin, ...params },
    timeout: 60000,
  });
  return res.data;
}

async function pricing(asin, params = {}) {
  // The pricing endpoint takes 'asin'. Sending 'query' here returns a validation error.
  const res = await axios.get(`${BASE}/amazon/pricing`, {
    headers,
    params: { asin, ...params },
    timeout: 60000,
  });
  return res.data;
}

(async () => {
  const page = await search('fitness tracker', { sort_by: 'featured' });

  const featured = page.products.filter((p) => p.is_sponsored);
  const organic = page.products.filter((p) => !p.is_sponsored);

  console.log(`${page.products_count} products, ${featured.length} featured, ${organic.length} organic`);

  featured.forEach((p) => {
    console.log(`  ad  slot ${p.sponsored_position}  ${p.price} ${p.currency}  ${p.title.slice(0, 60)}`);
  });

  organic.slice(0, 5).forEach((p) => {
    const badge = p.is_amazons_choice ? 'choice' : p.best_seller ? 'bestseller' : '';
    console.log(`  org rank ${p.organic_position}  ${p.price} ${p.currency}  ${badge}`);
  });

  if (organic.length) {
    const asin = organic[0].asin;
    const detail = await product(asin);
    console.log(`\n${detail.brand} / ${detail.price} ${detail.currency} / ${detail.rating} stars`);
    const offers = await pricing(asin);
    offers.pricing.forEach((o) => console.log(`  offer ${o.seller}  ${o.price}  ${o.condition}`));
  }
})();
